[BUGFIX] Fixes funkVis & grig.audio memory leak

FuroYT opened the pull request:
## Description
This Pull Request Fixes Memory Leaks related to funkVis and grig.audio due to them allocating a new array every frame instead of using the same array being cached and changing values

This PR depends on 2 other pr's on funkVis and grig.audio in the FunkinCrew org (linked below)
funkVis PR: https://github.com/FunkinCrew/funkVis/pull/16
grig.audio PR: https://github.com/FunkinCrew/grig.audio/pull/2 (uses the `refactor/fft-cam-version` branch instead of `funkin` due to changes in funkVis)
(these pr fixes the memory leak at its core too)

## Screenshots/Videos
https://github.com/user-attachments/assets/943a6ed0-4e43-4f75-8f6e-5e778aa84033
(notice how the memory doesn't leak anymore when previewing pico in the character selector)

## Notice
The libraries has to be updated in `hmm.json` so it works as expected

Hundrec reviewed (changes requested):
The memory leaks are fixed by this, but the visualizer now leans too far to treble and barely displays any bass.
Comparing the Character Select visualizer between this PR and 0.7.5 makes this clear.

FuroYT commented:
> The memory leaks are fixed by this, but the visualizer now leans too far to treble and barely displays any bass.
> Comparing the Character Select visualizer between this PR and 0.7.5 makes this clear.

I think this is due the grig audio branch being changed by ninjamuffin as I didn't change anything in there I just added caching arrays,

Despite this I'll take a look on it.

Hundrec commented:
Looks like your changes were very intertwined with Cam's! 

What I've done:
1. Based your PRs on new branches without Cam's commits that caused the incorrect visualizer behavior
2. Spent some time resolving conflicts after undoing Cam's commits from your funkVis/grig.audio branches
3. Added back the `var xscale = new Array<Float>();` on line 27 of `FFTVisualization.hx` because the game wouldn't compile without it
4. Tested Character Select and in-song visualizers. Character Select seems to leak a little now (probably because of the `xscale` line), while in-song memory doesn't climb.

What you should do now:
1. Test the current state of your PRs
2. Fix up the Character Select memory leak
3. Let me know when you're done and it's ready to test

Thanks!

FuroYT commented:
Hello Hundrec,

I've fixed the respective PR's to the fixes i previously added and added some more just in case
The game should compile with the `xscale` correctly moved to static instead of being a instance variable,

The pull request is now ready to be tested again.

Hundrec reviewed (approved):
Excellent work!
